### PR TITLE
Housekeeping! (clean samples_split_option, clean local val and test sets, misc. minor improvements)

### PR DESCRIPTION
**Pull request by bowni**
The way `samples_split_option` worked wasn't very clean, as it could take different types of value depending on being initialized to a basic split approach (`'random'` or `'stratified'`) or to an advanced approach (`[[7, 'specific'], [3, 'shared'], [2, 'shared']]`).

Now it looks better:

- it has to be declared as an array with the first value being the type (`'basic'` or `'advanced'`) and the second value the actual split chosen
- the test in `utils.py` for the scenarios combination is done on the type; same for the test in `main.py` for which data splitting function to call
- --> it makes it clean and easy to add new split approaches and even maybe new **types of approaches**

---

Also as suggested in #168, some cleaning with how local val and test sets are populated and used:

- The creation of local val sets is removed from the `preprocess_scenarios_data()` function
- The unused config parameter `single_partner_test_mode` is removed
- Attributes of object `Partner` for local val and test sets are kept
- Creating these val and test sets is now part of the data split functions. For the moment this is basically a `train_test_split()` on each partner's train set

---

Additionnally:

- add a few comments for readibility
- addresses #126

### CODE REVIEW

**Comment by RomainGoussault** [comment]
It's strange, it looks like the travis build was not automatically lauched for this PR ?

File: scenario.py [hunk 1]
```diff
@@ -587,8 +591,8 @@ def to_dataframe(self):
         dict_results["train_data_samples_count"] = len(self.dataset.x_train)
         dict_results["test_data_samples_count"] = len(self.dataset.x_test)
         dict_results["partners_count"] = self.partners_count
-        dict_results["amounts_per_partner"] = self.amounts_per_partner
-        dict_results["samples_split_option"] = self.samples_split_option
+        dict_results["dataset_fraction_per_partner"] = self.amounts_per_partner
+        dict_results["samples_split_description"] = self.samples_split_description
```
Comment (RomainGoussault) [inline scenario.py:595]: 👍

File: main.py [hunk 2]
```diff
@@ -137,7 +137,7 @@ def validate_scenario_list(scenario_params_list, experiment_path):
         current_scenario = scenario.Scenario(scenario_params, experiment_path, is_dry_run=True)
         current_scenario.instantiate_scenario_partners()
 
-        if isinstance(current_scenario.samples_split_option, list):
+        if isinstance(current_scenario.samples_split_description, list):
```
Comment (RomainGoussault) [inline main.py:140]: `if self.samples_split_type == "advanced"`
(like you did in `run_scenario()`)

Comment (bowni) [reply main.py:140]: Oh, I missed that one

**Pull request merged**